grpc/src/main/java/io/stargate/grpc/service/GrpcService.java

```diff
@@ -70,6 +71,49 @@ public void executeBatch(Batch batch, StreamObserver<Response> responseObserver)
     new BatchHandler(batch, CONNECTION_KEY.get(), persistence, responseObserver).handle();
   }
 
+  @Override
+  public StreamObserver<Query> executeQueryStream(StreamObserver<Response> responseObserver) {
+    return new StreamObserver<Query>() {
+      final AtomicLong inFlight = new AtomicLong(0);
+
+      @Override
+      public void onNext(Query query) {
+        inFlight.incrementAndGet();
+        System.out.println(
+            "Server side onNext for: " + query + " thread: " + Thread.currentThread().getName());
+        new StreamingQueryHandler(
+                query,
+                CONNECTION_KEY.get(),
+                persistence,
+                executor,
+                schemaAgreementRetries,
+                responseObserver,
+                inFlight)
+            .handle();
+      }
+
+      @Override
+      public void onError(Throwable t) {
+        new ExceptionHandler(responseObserver).handleException(t);
+      }
+
+      @Override
+      public void onCompleted() {
+        System.out.println(
+            "onCompleted server side "
+                + " thread: "
+                + Thread.currentThread().getName()
+                + " inFlight: "
+                + inFlight.get());
+        while (inFlight.get() > 0) {
```

Comment:
java-driver reactive support checks if all emitted events were processed, and only then sends the `onComplete`:
https://github.com/datastax/java-driver/blob/b7e384ea2500c1d0ef9cc882604128e6a486b2da/core/src/main/java/com/datastax/dse/driver/internal/core/cql/reactive/ReactiveResultSetSubscription.java#L239-L240

Reply:
According to reactive specification:
```
An Observable may emit no items at all. An Observable may also never terminate with either an OnCompleted or an OnError notification. That is to say that it is proper for an Observable to issue no notifications, to issue only an OnCompleted or an OnError notification, or to issue only OnNext notifications.
```
[The Contract Governing Notifications
](https://reactivex.io/documentation/contract.html)
So we may consider not supporting `onComplete` from our side and have never terminating `StreamObserver`